.idea/.gitignore

```diff
@@ -1,3 +1,73 @@
 # Default ignored files
 /shelf/
 /workspace.xml
+# IntelliJ IDEA modules and interim config files.
+*.iml
+.idea/*.xml
+.idea/.name
+.idea/artifacts
+.idea/libraries
+.idea/modules
+.idea/shelf
+
+# Do not ignore the following IDEA settings
+!.idea/misc.xml
+!.idea/codeStyleSettings.xml
+!.idea/codeStyles/
+!.idea/copyright/
+
+# Gradle interim configs
+**/.gradle/**
+
+# Generated source code
+**/generated/**
+
+# Gradle build files
+**/build/**
+
+# Build files produced by the IDE
+**/out/**
+
+# Ignore Gradle GUI config
+gradle-app.setting
+
+# Avoid ignoring Gradle wrapper jar file (.jar files are usually ignored)
+!gradle-wrapper.jar
+
+# Cache of project
+.gradletasknamecache
+
```

Comment:
Why do we need so much empty space? Are we reserving space for planting potato or what?